### PR TITLE
Do not forward connection-specific headers to/from H1 backend for H2 clients

### PR DESCRIPTION
**Pull request by dmpetroff**
For problem description see #1805
Implementation:
- for incoming connection headers I've changed parser to behave like it does for `Connection` header
- for connection headers coming from upstream, I'm setting `TFW_STR_HBH_HDR` header during parse phase

Related PR for tempesta-test is here: https://github.com/tempesta-tech/tempesta-test/pull/462
All that tests are passing locally

### CODE REVIEW

File: fw/http_parser.c [hunk 1]
```diff
@@ -9558,6 +9568,13 @@ tfw_h2_parse_req_hdr(unsigned char *data, unsigned long len, TfwHttpReq *req,
 				__FSM_H2_FIN(Req_HdrUser_AgentV, 10,
 					     TFW_TAG_HDR_USER_AGENT);
 			__FSM_H2_OTHER_n(4);
+		/* upgrade */
+		case TFW_CHAR4_INT('u', 'p', 'g', 'r'):
+			if (unlikely(!__data_available(p, 7)))
+				__FSM_H2_NEXT_n(Req_HdrUpgr, 4);
+			if (C4_INT(p + 3, 'r', 'a', 'd', 'e'))
+				__FSM_H2_DROP(Req_HdrUpgrade);
+			__FSM_H2_OTHER_n(4);
```
Comment (krizhanovsky) [inline fw/http_parser.c:9577]: Could you please add a comment with reference to RFC 9113 8.2.2 that we tread an HTTP/2 message having `Upgrade` header as malformed? The same for the `Proxy-Connection` above

File: fw/http_parser.c [hunk 2]
```diff
@@ -11761,19 +11817,29 @@ tfw_http_parse_resp(void *resp_data, unsigned char *data, unsigned int len,
...
+					/* Proxy-Connection */
+					__msg_hdr_chunk_fixup(data, __data_off(p + 16));
+					parser->_i_st = &&RGen_HdrOtherV;
+					parser->hdr.flags |= TFW_STR_HBH_HDR;
+					p += 16;
+					__FSM_MOVE_hdr_fixup(RGen_LWS, 1);
+				}
+				if (__data_available(p, 19)
+					&& C8_INT_LCM(p + 5, '-', 'a', 'u', 't', 'h', 'e', 'n', 't')
+					&& C8_INT7_LCM(p + 11, 'n', 't', 'i', 'c', 'a', 't', 'e', ':'))
```
Comment (krizhanovsky) [inline fw/http_parser.c:11835]: Please adjust the too wide lines here and above - we fit 80, maybe 85 if necessary, characters in line

**Review by krizhanovsky — APPROVED** [review]
LGTM with the couple of adjustments

**Review by const-t — CHANGES_REQUESTED** [review]
Current PR is good for me. But would be better to add the last connection-specific header `TE` to this PR. Tests for this header implemented in [RP 461 malformed](https://github.com/tempesta-tech/tempesta-test/pull/461)

File: fw/http_parser.c [hunk 3]
```diff
@@ -9807,6 +9845,11 @@ tfw_h2_parse_req_hdr(unsigned char *data, unsigned long len, TfwHttpReq *req,
 			     __h2_req_parse_x_forwarded_for,
 			     TFW_HTTP_HDR_X_FORWARDED_FOR, 0);
 
+	/* 'te' is read, process field-value */
+	TFW_H2_PARSE_HDR_VAL(Req_HdrTeV, msg,
+			     __h2_req_parse_te,
+			     TFW_HTTP_HDR_RAW, 1);
```
Comment (krizhanovsky) [inline fw/http_parser.c:9851]: Please add unit tests for the new parsing logic to `fw/t/unit/test_http2_parser.c`

**Review by krizhanovsky — CHANGES_REQUESTED** [review]
Unit tests are required

**Review by const-t — CHANGES_REQUESTED** [review]

**Review by krizhanovsky — APPROVED** [review]
LGTM. Couple of reviews ago I added a request to add a comment referencing the RFC, which was not addressed, so I just pushed the commit on my own. Please address **all** review comments in future (either comment them if you disagree or just fix).

**Review by const-t — APPROVED** [review]
LGTM

**Pull request merged**